src/abstract/Configurable.ts

```diff
@@ -0,0 +1,30 @@
+export abstract class Configurable<ConfigType extends object = {}> {
+    private readonly config: ConfigType;
```

Comment:
issue: I feel like we should just make this a protected member of `InternalPlugin`. I don't think it adds value to the non-plugin classes.
```suggestion
    protected config: ConfigType;
```

Reply:
the config should be read only and immuttable

there are many issues r8ght now where changes in one place to a config affect everythkng else (which is hard to test and maintain)

Reply:
Also config _is_ used in many of the non-plugin files